Model/WebApi/Addresscheck.php

```diff
@@ -0,0 +1,195 @@
+        \Magento\Checkout\Model\Session $checkoutSession
+    ) {
+        $this->addresscheck = $addresscheck;
+        $this->responseFactory = $responseFactory;
+        $this->checkoutSession = $checkoutSession;
+    }
+
+    /**
+     * Generate the confirm message from the given address
+     *
+     * @param  \Magento\Quote\Api\Data\AddressInterface $addressData
+     * @return string
+     */
+    protected function getConfirmMessage(\Magento\Quote\Api\Data\AddressInterface $addressData)
+    {
+        $sMessage  = __('Address corrected. Please confirm.')."\n\n";
+        $sMessage .= $addressData->getFirstname().' '.$addressData->getLastname()."\n";
+
+        $mStreet = $addressData->getStreet();
+        if (is_array($mStreet)) { // address can be string
+            $sMessage .= $mStreet[0]."\n"; // add first line of address array
+        } else {
+            $sMessage .= $mStreet."\n"; // add string directly
+        }
+        $sMessage .= $addressData->getPostcode().' '.$addressData->getCity();
+
+        return $sMessage;
+    }
+
+    /**
+     * Add the score to the correct session variable
+     *
+     * @param  \Magento\Quote\Api\Data\AddressInterface $oAddress
+     * @param  bool                                     $blIsBillingAddress
+     * @return void
+     */
+    protected function addScoreToSession(\Magento\Quote\Api\Data\AddressInterface $oAddress, $blIsBillingAddress)
+    {
+        $sScore = $this->addresscheck->getScore($oAddress);
+        if ($blIsBillingAddress === true) { // is billing address?
+            $this->checkoutSession->setPayoneBillingAddresscheckScore($sScore);
+        } else {
+            $this->checkoutSession->setPayoneShippingAddresscheckScore($sScore);
+        }
+    }
+
+    /**
+     * Set error message if checkout is configured to stop on error or set success = true instead
+     *
+     * @param  AddresscheckResponse $oResponse
+     * @return AddresscheckResponse
+     */
+    protected function handleErrorCase(AddresscheckResponse $oResponse)
+    {
+        $sHandleError = $this->addresscheck->getConfigParam('handle_response_error');
+        if ($sHandleError == 'stop_checkout') {
+            $oResponse->setData('errormessage', __($this->addresscheck->getErrorMessage())); // stop checkout with errormsg
+        } elseif ($sHandleError == 'continue_checkout') {
+            $oResponse->setData('success', true); // continue anyways
+        }
+        return $oResponse;
+    }
+
+    /**
+     * Handle the response according to its return status
+     *
+     * @param  AddresscheckResponse $oResponse
+     * @param  array                $aResponse
+     * @return AddresscheckResponse
+     */
+    protected function handleResponse(AddresscheckResponse $oResponse, $aResponse)
+    {
+        if ($aResponse['status'] == 'VALID') { // data was checked successfully
+            $oAddress = $this->addresscheck->correctAddress($oAddress);
```

Comment:
Bug here, `$oAddress` is not defined: https://scrutinizer-ci.com/g/PAYONE-GmbH/magento-2/inspections/93ca0b97-237d-4813-bf38-3c1592c5d997/issues/files/Model/WebApi/Addresscheck.php?status=new&selectedSeverities%5B0%5D=10&orderField=path&order=asc&honorSelectedPaths=0